app/src/main/java/ch/epfl/sdp/artificial_intelligence/Enemy.java

```diff
@@ -1,79 +1,127 @@
 package ch.epfl.sdp.artificial_intelligence;
 
-import java.util.ArrayList;
-import java.util.List;
-
 import ch.epfl.sdp.entity.EntityType;
 import ch.epfl.sdp.entity.Player;
 import ch.epfl.sdp.entity.PlayerManager;
-import ch.epfl.sdp.map.Displayable;
-import ch.epfl.sdp.map.GeoPoint;
-import ch.epfl.sdp.map.MapsActivity;
-
+import ch.epfl.sdp.game.GameThread;
+import ch.epfl.sdp.geometry.Area;
+import ch.epfl.sdp.geometry.LocalArea;
+import ch.epfl.sdp.geometry.Positionable;
+import ch.epfl.sdp.geometry.RectangleArea;
+import ch.epfl.sdp.geometry.UnboundedArea;
+import ch.epfl.sdp.geometry.Vector;
+
+/**
+ * Represents a hostile entity.
+ * The finite state machine is as follows:
+ * CHASE<------->PATROL+-----------+
```

Comment:
I'm not sure this graphic is understandable :). Maybe we should have a link to an image